test/standalone/dependencyFromBuildZig/build.zig

```diff
@@ -0,0 +1,16 @@
+const std = @import("std");
+
+pub fn build(b: *std.Build) void {
+    const test_step = b.step("test", "Test it");
+    b.default_step = test_step;
+
+    const dep1 = b.dependency("other", .{});
+
+    const build_runner = @import("root");
+    const deps = build_runner.dependencies;
+    const zon_decls = @typeInfo(deps.packages).Struct.decls;
+    const pkg = @field(deps.packages, zon_decls[0].name);
+    const dep2 = b.dependencyFromBuildZig(pkg.build_zig, .{});
```

Comment:
The current reflection code assumes that the generated `dependencies.zig` will only list one dependency and thus only works when this test is run as the build root. It doesn't work in the context of the compiler tests because there it's a dependency of `standalone_test_cases`.

Replacing it with
```suggestion
    const dep2 = b.dependencyFromBuildZig(@import("other"), .{});
```
should fix it.